src/libostree/ostree-bootconfig-parser.c

```diff
@@ -56,6 +59,96 @@ ostree_bootconfig_parser_clone (OstreeBootconfigParser *self)
   return parser;
 }
 
+/**
+ * ostree_bootconfig_parser_parse_tries:
+ * @self: Parser
+ * @path: File path
+ *
+ * Parses a suffix of two counters in the form "+LEFT-DONE" from the end of the
+ * filename (excluding file extension).
+ */
+static void
+parse_bootloader_tries (const char *filename, guint64 *out_left, guint64 *out_done, GError **error)
+{
+  gchar *counter = NULL;
+  gchar *old_counter = NULL;
+  *out_left = 0;
+  *out_done = 0;
+
+  for (;;)
+    {
+      char *plus = strchr (counter ?: filename, '+');
+      if (plus)
+        {
+          /* Look for the last "+" symbol in the filename */
+          counter = plus + 1;
+          continue;
+        }
+      if (counter)
+        break;
+
+      /* No boot counter found */
+      return;
+    }
```

Comment:
I had to stare at this a bit...I think it's correct as is, but wouldn't it be a lot clearer to use `strrchr ()` here (note the extra `r` to search from the end)?

i.e.
```
const char *counter = strrchr (filename, '+');
if (!counter)
  return;
counter += 1;
```

Reply:
fixed, thanks!